spark/src/main/scala/geotrellis/spark/clip/ClipToGrid.scala

```diff
@@ -16,191 +16,111 @@
 
 package geotrellis.spark.clip
 
-import geotrellis.raster.GridBounds
 import geotrellis.spark._
 import geotrellis.spark.tiling._
-import geotrellis.util._
 import geotrellis.vector._
 
-import com.vividsolutions.jts.geom.prep.{PreparedGeometry, PreparedGeometryFactory}
+import com.vividsolutions.jts.geom.prep.PreparedGeometryFactory
 import org.apache.spark.rdd._
+import scala.util.Try
 
 object ClipToGrid {
-  /** Trait which contains methods to be used in determining
-    * the most optimal way to clip geometries and features
-    * for ClipToGrid methods.
-    */
-  trait Predicates {
-    /** Returns true if the feature geometry covers the passed in extent */
-    def covers(e: Extent): Boolean
-    /** Returns true if the feature geometry is covered bythe passed in extent */
-    def coveredBy(e: Extent): Boolean
-  }
-
-  /** Clips a feature to the given extent, that uses the given predicates
-    * to avoid doing intersections where unnecessary.
-    */
-  def clipFeatureToExtent[G <: Geometry, D](
-    e: Extent,
-    feature: Feature[G, D],
-    preds: Predicates
-  ): Option[Feature[Geometry, D]] =
-    if(preds.covers(e)) { Some(Feature(e, feature.data)) }
-    else if(preds.coveredBy(e)) { Some(feature) }
-    else {
-      feature.geom.intersection(e).toGeometry.map { g =>
-        Feature(g, feature.data)
-      }
-    }
 
   /** Clip each geometry in the RDD to the set of SpatialKeys
     * which intersect it, where the SpatialKeys map to the
     * given [[LayoutDefinition]].
     */
-  def apply[G <: Geometry](
-    rdd: RDD[G],
-    layout: LayoutDefinition
-  )(implicit d: DummyImplicit): RDD[(SpatialKey, Geometry)] =
-    apply[G, Unit](rdd.map(Feature(_, ())), layout)
-      .mapValues(_.geom)
+  def apply[G <: Geometry](layout: LayoutDefinition, rdd: RDD[G]): RDD[(SpatialKey, Geometry)] =
```

Comment:
The argument ordering is intentional and fits with the rest of the method extensions across the library.

the `object Verb` has `apply` methods that take as a first argument the subject. That subject is the `self` in the MethodExtensions.